mypy/checker.py

```diff
@@ -1360,16 +1360,22 @@ def check_method_override_for_base_with_name(
             # it can be checked for compatibility.
             original_type = base_attr.type
             original_node = base_attr.node
+            if isinstance(original_node, TypeAlias):
```

Comment:
Should we reject other kinds of nodes as well, such as `TypeInfo`, `TypeVarExpr` or module reference? Some of these could cause crashes as well. It would be good to fix other related issues at the same time. 

I think that it would be best to check all combinations of base class and subclass definitions, such as 1) type alias 2) class 3) function 4) variable 5) type variable 6) module reference 7) other magic assignments (NewType or NamedTuple or TypedDict). Some of these are already tested, so they can be left out (like function vs function).